### ISSUE TITLE
Crown on the top of the current best player

### ISSUE THREAD

**Issue by matfrem** [UltraStar-Deluxe/Play#233]
Issue type: Feature request

### Actual behaviour

When playing two players, we have to read the score to find the best player.

### Expected behaviour

When playing two players, it's always fun to see this as a race. It would be nice to have a crown on the top of the best player, in real-time while singing.

![sing star](https://bilder.buecher.de/zusatz/57/57621/57621080_deta_1.jpg)
![just dance](https://i.ytimg.com/vi/ycj4e3TknHU/maxresdefault.jpg)

As a first contribution, I'd like to add this to the current codebase.

**Comment by achimmihca** [comment]
Nice idea.
I would change it to: "When playing with more than one player..."
This feature works just as well when playing with more than two players.

**Comment by achimmihca** [comment]
For the implementation, I would

- extend Assets/Scenes/Sing/PlayerController/PlayerUiController/PlayerUi.prefab with a crown image. This way, every Player UI has this crown
- add a script to this crown image, with show/hide method (e.g. gameObject.SetActive(false / true)). This way the PlayerUiController can hold a reference to the object via Inject(searchMethod = SearchMethod.GetComponentInChildren).
- in SingSceneController, you can iterate the PlayerControllers and thus the PlayerUiController and PlayerScoreController.
    - Subscribe to NoteScoreEventStream and SentenceScoreEventStream of the PlayerScoreControllers
    - in callback, sort them by TotalScore.
    - then show the crown of the top player and hide it if it was shown on another player before.

**Comment by matfrem** [comment]
Thanks,
In PlayerUiController , would it be more like
**public** PlayerCrown playerCrown; to change crown status directly,
or private, and a method in PlayerUiController  to change crown status ?

**Comment by achimmihca** [comment]
I recommend 
`public PlayerUiController PlayerUiController { get; private set; }`
(C# short syntax for getter / setter with [backing field](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/properties#properties-with-backing-fields))

**Comment by matfrem** [comment]
Ok it works nicely "as-is", thanks ! 
_But_
Now I'd like to add a starParticle effect when crown gets displayed.
But this StarParticle class is in the "Scenes" module, and my PlayerCrown class is in the "Common" module.

So as does "TotalScoreDisplayer", I decided to create a "PlayerCrownDisplayer" in "Scenes" instead of "PlayerCrown" in"Common" modules.
But I feel that prevents me from using depndency injection in SingSceneController (as object is not in children), so I went back to some code like "playerController.**PlayerUiController**.PlayerCrownDisplayer.ShowCrown(false);" in SingSceneController... It works... not sure it's the best thing to do.
